variables.mk

```diff
@@ -125,10 +125,9 @@ HARNESS_SMEMS_FILE ?= $(build_dir)/$(long_name).harness.mems.v
 HARNESS_SMEMS_CONF ?= $(build_dir)/$(long_name).harness.mems.conf
 HARNESS_SMEMS_FIR  ?= $(build_dir)/$(long_name).harness.mems.fir
 
-sim_files                  ?= $(build_dir)/sim_files.f
-sim_top_blackboxes         ?= $(build_dir)/firrtl_black_box_resource_files.top.f
-sim_harness_blackboxes     ?= $(build_dir)/firrtl_black_box_resource_files.harness.f
-sim_common_files           ?= $(build_dir)/sim_files.common.f
+sim_files          ?= $(build_dir)/sim_files.f
+sim_blackboxes     ?= $(build_dir)/firrtl_black_box_resource_files.sim.f
+sim_common_files   ?= $(build_dir)/sim_files.common.f
```

Comment:
Don't we still need a harness blackbox file? E.g. with a hbwif module you still have non-SRAM blackboxes in the harness.

Reply:
My impression is that they would just be added in the `sim.f` file. The current `blackbox.f` file seems to match the `sim.f` version, so I think they are similar if not the same.

Reply:
It's not clear to me which file is `sim.f` because there are multiple files ending in `.f` that contain `sim`. `firrtl_black_box_resource_files.sim.f` ?

What is the difference between `$(sim_files)` and `$(sim_blackboxes)` ?

Reply:
Sorry. By `sim.f` I mean `.sim.f`. So I am referring to the `firrtl_black_box_resource_files.sim.f`. 

The difference between the two is that the `sim_blackboxes` is the file generated by the RTL during elaboration/firrtl. The `sim_files` is generated by the `GenerateSimFiles` step (run before elaboration). It adds all necessary files needed for either Verilator or VCS in the `sim_files.f`.

However, these two files (`sim_blackboxes` and `sim_files`) can conflict so thats why we need the dedup step.

Reply:
So what lives in `$(sim_common_files)` then? Is it things that are common to both the top and harness or things that are in both `$(sim_files)` and `$(sim_blackboxes)`?

We separate out the top and harness blackboxes because we pass (some of) the top blackboxes to synthesis. Depending on the synthesis settings used, the hierarchy may be preserved and there would be module conflicts between the modules in the top blackbox .f list and the post-synthesis netlist, which is why we want to have a separate harness .f.

Reply:
`$(sim_common_files)` is the all unique files needed for simulation in VCS or Verilator (`.cc`, `.v` files). So it is the second option you listed... things in both `$(sim_files)` and `$(sim_blackboxes)`.

As for the separation, I think that needs to be discussed in the corresponding BARSTOOLS PR.

Reply:
@jwright6323 Is there a problem if we pass all of the verilog files in the `.f` files into synthesis? If they aren't instantiated the tools shouldn't have a problem (and if they do we can add appropriate `ifdef SYNTHESIS`).

Reply:
@colinschmidt I think it's probably ok to pass them all to synthesis, but @albert-magyar highlighted the actual problem- we need a version of the .f for things that exist _only_ in the harness so that we can avoid conflicts with the post-syn netlist.

Reply:
I forgot to circle back to this comment, but this generates two .f files once again.